src/SW_Weather.c

```diff
@@ -1662,30 +1697,27 @@ void _read_weather_hist(
 
         // Calculate average air temperature if min/max not missing
         if (!missing(weathInput[dailyInputIndices[TEMP_MAX]]) &&
-            !missing(weathInput[dailyInputIndices[TEMP_MIN]])) {
+                            !missing(weathInput[dailyInputIndices[TEMP_MIN]])) {
 
-          yearWeather->temp_avg[doy] = (weathInput[dailyInputIndices[TEMP_MAX]] +
-                                        weathInput[dailyInputIndices[TEMP_MIN]]) / 2.0;
+              yearWeather->temp_avg[doy] = (weathInput[dailyInputIndices[TEMP_MAX]] +
+                                            weathInput[dailyInputIndices[TEMP_MIN]]) / 2.0;
         }
 
-        if(!use_cloudCoverMonthly) {
-            if(dailyInputFlags[CLOUD_COV]) {
-                yearWeather->cloudcov_daily[doy] = weathInput[dailyInputIndices[CLOUD_COV]];
-            }
+        if(dailyInputFlags[CLOUD_COV]) {
+            yearWeather->cloudcov_daily[doy] = weathInput[dailyInputIndices[CLOUD_COV]];
         }
 
-        if(!use_windSpeedMonthly) {
-            if(dailyInputFlags[WIND_SPEED]) {
-                yearWeather->windspeed_daily[doy] = weathInput[dailyInputIndices[WIND_SPEED]];
-            } else if(dailyInputFlags[WIND_EAST] && dailyInputFlags[WIND_NORTH]) {
-                yearWeather->windspeed_daily[doy] = sqrt(squared(weathInput[dailyInputIndices[WIND_EAST]]) +
-                                                         squared(weathInput[dailyInputIndices[WIND_NORTH]]));
-            }
+        if(dailyInputFlags[WIND_SPEED]) {
+            yearWeather->windspeed_daily[doy] = weathInput[dailyInputIndices[WIND_SPEED]];
+
+        } else if(dailyInputFlags[WIND_EAST] && dailyInputFlags[WIND_NORTH]) {

```

Comment:
Considering `_read_weather_hist()` a bit further: we need to capture even more cases of missing values and propagate them through to `allHist`. If we just go ahead and calculate with SW_MISSING (which is just 999) then we produce a value that is not SW_MISSING), e.g., `500.5 = (2 + SW_MISSING) / 2)` which is a nonsense value. In order for `SW_WTH_new_day()` to actually be able to fail if a value is missing, we need to propagate all missing input values. Thus, the following cases must additionally propagate SW_MISSING (such as for instance `yearWeather->temp_avg[doy] = ...` is doing):
- wind speed calculated from east/north wind components
- relative humidity calculated from min/max relative humidity
- relative humidity calculated from specific humidity
- vapor pressure calculated from dewpoint temperature
- vapor pressure calculated from min/max relative humidity
- vapor pressure calculated from relative humidity

Cases where we just copy input values into our daily data structure don't need to be handled separately, e.g., relative humidity copied from input relative humidity (case `dailyInputIndices[REL_HUMID]`).